src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -705,11 +705,8 @@ private Expr CreateArgumentEXPR(ArgumentObject argument, LocalVariableSymbol loc
             // as well so that overload resolution can find them.
             if (callingType.IsWindowsRuntimeType())
             {
-                TypeArray collectioniFaces = callingType.GetWinRTCollectionIfacesAll(SymbolLoader);
```

Comment:
It also tries to pretend the array is immutable - it has getter-only indexer, but it also has "Items" so it is not entirely safe.

Perhaps the type must be a struct.